tests/integration_test/iam_imds_data_path_test/iam_imds_data_path_test.go

```diff
@@ -0,0 +1,110 @@
+package iam_imds_data_path_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/esp-v2/tests/endpoints/echo/client"
+	"github.com/GoogleCloudPlatform/esp-v2/tests/env"
+	"github.com/GoogleCloudPlatform/esp-v2/tests/env/platform"
+	"github.com/GoogleCloudPlatform/esp-v2/tests/utils"
+
+	comp "github.com/GoogleCloudPlatform/esp-v2/tests/env/components"
+)
+
+var testDataPathArgs = []string{
+	"--service_config_id=test-config-id",
+	"--rollout_strategy=fixed",
+	"--backend_dns_lookup_family=v4only",
+	"--suppress_envoy_headers",
+}
+
+func TestDataPathImdsSuccessWhenIamDown(t *testing.T) {
+	s := env.NewTestEnv(comp.TestDataPathImdsSuccessWhenIamDown, platform.EchoRemote)
+
+	// Simulate Iam failures.
+	s.SetIamResps(map[string]string{}, 100)
```

Comment:
could you add comment on how IAM or Imds is configured in this test?  what configure determine IAM or Imds for service_control and backend_auth?

Reply:
Also don't undersand why IAM is down, but request still works.

Reply:
I re-wrote the integration test with more fields and test cases. Maybe this will be clearer?